# Invalid CNPJ or CPF

`cnpj_cpf` is the column identifying the company or individual who received the payment made by the congressperson. Having this value empty should mean that it's an expense made outside Brazil, with a company (or person) without a Brazilian ID.

In [1]:
import numpy as np
import pandas as pd

dataset = pd.read_csv('../data/2016-11-19-reimbursements.xz',
                      dtype={'applicant_id': np.str,
                             'cnpj_cpf': np.str,
                             'congressperson_id': np.str,
                             'subquota_number': np.str},
                      low_memory=False)
dataset.shape

(1532491, 31)

In [2]:
from pycpfcnpj import cpfcnpj

def validate_cnpj_cpf(cnpj_or_cpf):
    return (cnpj_or_cpf == None) | cpfcnpj.validate(cnpj_or_cpf)



cnpj_cpf_list = dataset['cnpj_cpf'].astype(np.str).replace('nan', None)
dataset['valid_cnpj_cpf'] = np.vectorize(validate_cnpj_cpf)(cnpj_cpf_list)

In [3]:
keys = ['year',
        'applicant_id',
        'document_id',
        'total_net_value',
        'cnpj_cpf',
        'supplier']
dataset.loc[~dataset['valid_cnpj_cpf'], keys].head()

year applicant_id  document_id  total_net_value     cnpj_cpf  \
53466  2009         1607      1748889           123.57  11111111111   
53467  2009         1607      1748896           100.25  11111111111   
53468  2009         1607      1748909           229.25  11111111111   
53469  2009         1607      1748911            18.89  11111111111   
53470  2009         1607      1748915           581.85  11111111111   

                         supplier  
53466                    CAP HORN  
53467                    CAP HORN  
53468  DENSKALDEDEKOK  RESTAURANT  
53469                BELLA CENTER  
53470      FIRST HOTEL SKT. PETRI

With 1,532,491 records in the dataset and just 5 with invalid CNPJ/CPF, we can probably assume that the Chamber of Deputies has a validation in the tool where the congressperson requests for reimbursements. These represent a mistake in the implemented algorithm.